.github/workflows/tics.yaml

```diff
@@ -0,0 +1,61 @@
+name: TICS
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Verify snap builds successfully
+        id: build
+        uses: canonical/action-build@v1
+
+      - name: Upload the built snap
+        uses: actions/upload-artifact@v4
+        with:
+          name: snap_artifact
+          path: ${{ steps.build.outputs.snap }}
+
+  tics-report:
+    needs: build
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          python -m pip install --upgrade pip
+          python -m pip install flake8
```

Comment:
Why this is needed? Can we install it with tox?

Reply:
> please ensure the agent has flake8, pylint and all python dependencies installed before the TiCS run

As per one of the security team members

Reply:
Can you provide some comment about why it's there. Thanks

Reply:
Its there

Reply:
uhm but shouldn't the tics action ensure its dependencies are installed? This sounds like a bug in the action (of course in the short term we should install the dependency ourselves). wdyt?

Reply:
I know, it's weird, but I talked with the folks from security team, and they told me to explicitly install the deps that the action misses

Reply:
But I'll file a bug upstream just in case

Reply:
can you please also link the bug in a code comment please?

Reply:
https://warthogs.atlassian.net/browse/TICSISSUES-3 dropping here for you to see